app/kuma-dp/cmd/run.go

```diff
@@ -477,7 +495,7 @@ func setupObservability(ctx context.Context, kumaSidecarConfiguration *types.Kum
 		tpEnabled,
 	)
 	metricsServer := metrics.New(
-		core_xds.MetricsHijackerSocketName(cfg.DataplaneRuntime.SocketDir, cfg.Dataplane.Name, cfg.Dataplane.Mesh),
+		core_xds.MetricsHijackerSocketName(cfg.DataplaneRuntime.WorkDir, cfg.Dataplane.Name, cfg.Dataplane.Mesh),
```

Comment:
Following the earlier comment:

```suggestion
		core_xds.MetricsHijackerSocketName(cfg.DataplaneRuntime.SocketDir, cfg.Dataplane.Name, cfg.Dataplane.Mesh),
```